### ISSUE TITLE
Feedback from @kfogel on if "sloppy" will put people off, if "archive" or "permanent" should be more explicit, and what the canonical reference for SLOPI is

### ISSUE THREAD

**Issue by pdurbin** [good-labs/slopi-communication#3]
Over at https://github.com/good-labs/good-labs.github.io/issues/23#issuecomment-491405176 @kfogel had the following feedback:

"I love that you've codified this, @pdurbin. I worry that the acronym will put some people off -- it's cute that it sounds like "sloppy", but it's also a disadvantage, because people don't like to be sloppy :-) -- but maybe I'm overthinking it and that won't be a problem in practice.

One quality is only included by implication: "Archive" (or "Permanent"). I'm not sure whether there's a way to fit that into the acronym, or to make a new acronym... again, it's implied anyway, so perhaps "SLOPI" is the way to go.

Please consider writing a blog post about this, if you have the time, so that when people introduce the term "SLOPI" into a conversation they have a canonical reference they can point others to."

The feedback feels like it deserves its own issue. I'm certainly ok with brainstorming on a better acronym. I'm definitely open to ideas!

**Comment by pdurbin** [comment]
My first thought is that we all know that naming is hard. Check out this tweet from https://twitter.com/stewart/status/780906639301812225 about the naming of Slack. Today I learned that like me Stewart Butterfield was playing around with acronyms when he came up with the name for Slack.  On Nov 12, 2012 he tossed out the phrase "Searchable Log of All Conversation & Knowledge" in the following exchange:

9:48 AM (stewart) I was up very late last night thinking about things

9:49 AM (stewart) and it is no exaggeration to say "WOW!"

9:49 AM (stewart) Also, I have a better code name

9:49 AM (stewart) "Slack" and/or "Slack App"

9:49 AM (stewart) Searchable Log of All Conversation & Knowledge

9:49 AM (stewart) but, "slack" is also just nice to say

9:51 AM (eric) I like it, but... it has kind of negative connotations, too.

9:51 AM (eric) Our users would be Slackers :)

9:52 AM (stewart) ta da!

9:52 AM (stewart) just a codename

9:52 AM (eric) ah, cool

9:53 AM (stewart) but wait till I have Kuke draw you the "slack" (8 armed, fuzzy-headed, one-eyed, cute little thing that eats ALL your work communication and retains it for you in a nice searchable form ... ie., poop)

Some observations for the exchange above:

- Slack is nice to say.
- Slack has negative connotations and this is seen as a positive. The users are slackers.
- A fun name can lead to a fun logo.

**Comment by pdurbin** [comment]
I think I'm going to respond to feedback from @kfogel bit by bit.

## Is "sloppy" an advantage or a disadvantage?

Karl's comment:

> "I love that you've codified this, @pdurbin. I worry that the acronym will put some people off -- it's cute that it sounds like "sloppy", but it's also a disadvantage, because people don't like to be sloppy :-) -- but maybe I'm overthinking it and that won't be a problem in practice."

My thoughts:

Pros:

- "Sloppy" is lighthearted and fun. Someday SLOPI could have a great logo.
- "Sloppy" is nice to say. (See https://github.com/good-labs/slopi-communication/issues/3#issuecomment-491504514 about how Slack was named and how it's also nice to say.) "Sloppy" rolls off the tongue.
- Being an adjective, "sloppy" can be applied well to nouns like "communication" or "issue tracker". "Our project has a SLOPI communication style." A noun like "slop" sounds more like a product name. We are defining a style, not a product. We want products or solutions to list "SLOPI" as a feature of their software. "Channels can be configured to be SLOPI (great for open source community building!) or private (great for discussion of security issues or code of conduct violations).

Cons:

- "Sloppy" has a negative connotation. Calling someone sloppy at work is an insult. "Slipshod" is even more insulting.
- "Sloppy" may not translate well into other languages.
- "Sloppy" doesn't lend itself toward what people call themselves the way that "Slack" leads to users calling themselves "Slackers". What do you call someone who likes to be sloppy? A slob? That's pretty negative. "Sloppy Joe" could work for men but excludes women, and it's an Americanism. I just learned that "Sloppy Giuseppe" is a thing: https://en.wikipedia.org/wiki/Sloppy_Giuseppe 😄 

What am I forgetting? What do people think about "sloppy"? 😄 Please feel free to give a 👍 on this comment if you like "sloppy" and a 👎 if you don't like "sloppy". 😄

**Comment by pdurbin** [comment]
## Does SLOPI have anything to say about "archive" or "permanent"?

Karl's comment:

> "One quality is only included by implication: "Archive" (or "Permanent"). I'm not sure whether there's a way to fit that into the acronym, or to make a new acronym... again, it's implied anyway, so perhaps "SLOPI" is the way to go."

I believe in archives, in libraries, in preservation. However, will our communication be preserved indefinitely? I sort of doubt it. Take a look at https://indieweb.org/site-deaths and you'll be reminded of many services you've heard of that are completely gone.

Who would be responsible for the archiving, the permanence? The open source project? Ideally, yes, the open source project would self host all communication and take great care in preserving it, or at least the important communication. However, projects often use free hosting from GitHub and other service providers without much thought of the service provider going dark some day.

So if Karl means "permanent until your provider goes out of business", sure I agree. Maybe we could add another "P" and make it "SLOPPI".

I often think about "Cool URIs don't change" at https://www.w3.org/Provider/Style/URI but a lot of people don't seem to care if URLs or whole domains go dark, go missing.

In short, archiving and permanence is a nice to have but I think it's too high a bar for SLOPI. I'm trying to get people to have more conversations in the open, in public, when it makes sense to. Archiving those conversations is an effort I can get behind, but I don't think I want to make adoption of a SLOPI communication style so onerous that adopters must become archivists. It's a tough and important job, a separate concern. That said, I've had pretty good luck with https://github.com/minrk/archive-gitter for backing up Gitter data, for data liberation of Gitter conversation. I don't know if there's a good way to back up Google Groups data. Again, ideally, projects are self hosting and controlling their own data but this is a lot of work and perhaps unrealistic for small projects.

**Comment by pdurbin** [comment]
## Canonical reference for SLOPI

Karl's comment:

> "Please consider writing a blog post about this, if you have the time, so that when people introduce the term "SLOPI" into a conversation they have a canonical reference they can point others to."

I don't actually have a blog. I do have a wiki and I added SLOPI to http://wiki.greptilian.com/ideas the other day and I linked to https://github.com/good-labs/slopi-communication (this repo) so I guess for now that's the canonical reference. 😄

**Comment by pdurbin** [comment]
I have a couple more thoughts on 'Does SLOPI have anything to say about "archive" or "permanent"?'

I opened https://discourse.opensourcedesign.net/t/downloading-a-copy-of-the-logs-for-opensourcedesign-from-botbot-me-before-its-shutdown/737/2 when BotBot.me was shut down: https://lincolnloop.com/blog/saying-goodbye-botbotme/ . I thought that perhaps others would see value in saving the chat logs. (I linked to them here and there over the years.) I guess not. Now they're gone. Links to that content are now dead. So it goes. If I stumble upon a link I made maybe I'll see if the Wayback Machine has the content.

@coding-horror said, "We switched away from Slack to the open source https://rocket.chat/  -- as usual (we've switched chat services 3+ times now) threw away all chat history and not a single thing of value was lost" at https://twitter.com/codinghorror/status/950843266231971840 . It's somewhat odd to me that Slack stands for "Searchable Log of All Conversation & Knowledge" but there's no value in that data? Just throw it all away? Is the same true of your issue tracker, your mailing list, your forum, your blog, your website? Where do you draw the line?

**Comment by pdurbin** [comment]
One more quick thought about the canonical reference for SLOPI is that while I'm not an academic, I'm surrounded by them at work and I wonder if I should attempt to publish a short paper with a DOI. After all, I was inspired by how "FAIR" took off as an acronym for Findable, Accessible, Interoperable, and Reusable when it comes to research data and that term is quite new. The paper was published in 2016 and can be found at https://doi.org/10.1038/sdata.2016.18 . So the question becomes if I care enough to attempt to publish something. Or do I just call https://github.com/good-labs/slopi-communication the canonical reference. Or just tweet about it and call that tweet the canonical reference. I don't know. 😄

**Comment by pdurbin** [comment]
I think I've responded enough here to comments by @kfogel to close this issue. Some stuff is still somewhat unresolved, but that's ok. Thanks again for the feedback!

**Comment by pdurbin** [comment]
@kfogel I finally got around to trying to write up a canonical reference for SLOPI. I started a blog in 2020 so I thought I'd put it there.

The title is "**Searchable Linkable Open Public Indexed (SLOPI) Communication or Why open source projects should avoid Slack**" and it can be found at http://blog.greptilian.com/2020/01/25/slopi-communication/

It's not perfect but I think it's ok and I'm trying to keep it short. Feedback is very welcome.

In 3c1861c linked to this new blog post from the README:

https://github.com/good-labs/slopi-communication/blob/master/README.md#canonical-reference-for-slopi

I hope this helps!
